drivers/clock_control/clock_control_mcux_ccm.c

```diff
@@ -327,6 +327,53 @@ static int mcux_ccm_get_subsys_rate(const struct device *dev,
 		*rate = CLOCK_GetClockRootFreq(kCLOCK_Flexspi2ClkRoot);
 		break;
 #endif
+
+#if CONFIG_MCUX_FLEXIO
+	case IMX_CCM_FLEXIO1_CLK:
+	{
+		uint32_t flexio_mux = CLOCK_GetMux(kCLOCK_Flexio1Mux);
+		uint32_t source_clk_freq = 0;
+
+		if (flexio_mux == 0) {
+			source_clk_freq = CLOCK_GetPllFreq(kCLOCK_PllAudio);
+		} else if (flexio_mux == 1) {
+			source_clk_freq = CLOCK_GetUsb1PfdFreq(kCLOCK_Pfd2);
+	#ifdef PLL_VIDEO_OFFSET /* fsl_clock.h */
+		} else if (flexio_mux == 2) {
+			source_clk_freq = CLOCK_GetPllFreq(kCLOCK_PllVideo);
+	#endif
+		} else {
+			source_clk_freq = CLOCK_GetPllFreq(kCLOCK_PllUsb1);
+		}
+
+		*rate = source_clk_freq / (CLOCK_GetDiv(kCLOCK_Flexio1PreDiv) + 1)
+					/ (CLOCK_GetDiv(kCLOCK_Flexio1Div) + 1);
+	} break;
+#endif
```

Comment:
Shouldn't the `#endif` here be after the case for `IMX_CCM_FLEXIO2_3_CLK`? Also, should we check if `DT_NODE_HAS_STATUS(DT_NODELABEL(flexio1), okay)` for the first case definition here?

Reply:
@danieldegrasse, I agree that there should be a check for `DT_NODE_HAS_STATUS(DT_NODELABEL(flexio1), okay)`, because `flexio1` is not always enabled. But `#endif`, it seems to me better not to change it, as is done in lines `214` and `221` of the same file.

Reply:
Yeah, you are right- using `&&CONFIG_MCUX_FLEXIO` with each conditional is best here